mbed.h

```diff
@@ -113,4 +113,6 @@ using namespace mbed;
 using namespace std;
 #endif
 
+MBED_USED int main(void);
```

Comment:
Sure. I was not sure about the correct place for this line, but I can see I forgot to ask.

Reply:
Ok, I tried with [mbed_boot_gcc_arm.c](https://github.com/ARMmbed/mbed-os/blob/249d35c873ff069a2eeccf0897df3ffae1997a74/rtos/source/TARGET_CORTEX/TOOLCHAIN_GCC_ARM/mbed_boot_gcc_arm.c#L73-L78) with no luck. I think the attribute has to be added in the same translation unit as the `main()` itself. How about [mbed_toolchain.h](https://github.com/ARMmbed/mbed-os/blob/249d35c873ff069a2eeccf0897df3ffae1997a74/platform/mbed_toolchain.h#L516), would that be acceptable?

Reply:
@kjbracey-arm, what do you think; is mbed_toolchain.h any better than mbed.h?

Reply:
Oh, sorry, missed this. (Then saw it, then forgot it).

I was more thinking it should be out of the way, and not visible to the app, because I'm nervous that it might cause build errors. A compiler can be a bit lax about accepting different `main` prototypes, and this locks us down to this (albeit correct) one, so it might make `void main(void)` become refused.

But if we have to have it visible at the definition of main, can't be avoided, I guess. mbed_toolchain.h is a bit better.

I'd still like to see some sort of alternative - eg can we stick a MBED_USED on wrap_main?

Reply:
> I'd still like to see some sort of alternative - eg can we stick a MBED_USED on wrap_main?

I've tried that, but couldn't make it work. Either a modified main.cpp or any header it includes -- only these fix the issue for me. :disappointed:

Reply:
Actually, there is a better solution -- adding the `"-u main"` to `"ld"` flags.